sequencer/src/catchup.rs

```diff
@@ -135,10 +135,16 @@ impl<ApiVer: StaticVersionType> StateCatchup for StatePeers<ApiVer> {
                 .send()
                 .await
             {
-                Ok(res) => match res.proof.verify(&fee_merkle_tree_root) {
-                    Ok(_) => return Ok(res),
-                    Err(err) => tracing::warn!("Error verifying account proof: {}", err),
-                },
+                Ok(res) => {
+                    if res.proof.account == account.into() {
```

Comment:
I think this would be clearer as early return:

```rust
if res.proof.account != account.into() {
  tracing::warn!("Invalid proof received from peer {:?}", client.url);
}
```

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/2091/commits/38d714e6170ce43614b753fbe8255af0ac6d5b7a